packages/data-provider/src/config/config.ts

```diff
@@ -48,6 +49,7 @@ export type AppConfig = {
 	limits: LimitsConfig;
 	logger: LoggerConfig;
 	schemaService: SchemaServiceConfig;
+	authMiddleware?: (req: Request, res: Response, next: NextFunction) => void;
```

Comment:
Applications that implements Lyric provider will pass a custom middleware for authentication